mautic/mautic
# Campaign preview

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: Branch adds a "preview" tab to the details view in campaignBundle. Tab content is a new preview view that is essentially a non-editable view of the campaign builder with slightly modified UI. The preview view should update in real time during any campaign builder updates, but the user should never be able to modify the campaign in any way from the preview view.

#### Steps to test this PR:
1. Campaigns > any campaign > Details
2. Scroll down to stats tabs
3. Observe "preview" tab - it should be automatically selected, and displaying the current campaign flowchart created in the builder
4. Details > Edit > Launch Campaign Builder
5. Modify campaign by adding/removing events
6. save/close campaign builder
7. Save edit view, close > Details
8. Observe "preview" tab - it should be updated with the latest changes from the campaign builder
9. Repeat steps 4-8 as necessary to confidently approve the feature
